src/azul/queues.py

```diff
@@ -235,15 +235,19 @@ def wait_to_stabilize(self) -> int:
             logger.info('Message count history (most recent first) is %r.',
                         list(reversed(total_lengths)))
 
-            if len(total_lengths) == total_lengths.maxlen:
-                cummdiff = sum(
-                    abs(first - second)
-                    for first, second in more_itertools.pairwise(total_lengths)
-                )
-                if cummdiff == 0:
-                    final_length = total_lengths[-1]
-                    logger.info('The queues have stabilized.')
-                    break
+            min_iter = 60 // sleep_time
+            total_lengths_len = len(total_lengths)
+            if total_lengths_len >= min_iter:
+                is_zeros = set(list(total_lengths)[-min_iter:]) == {0, }
+                if is_zeros or total_lengths_len == total_lengths.maxlen:
+                    cummdiff = 0 if is_zeros else sum(
```

Comment:
It's confusing to force cummdiff to 0 when really it isn't, only to trip a subsequent conditional.